src/fitnesse/slim/fixtureInteraction/DefaultInteraction.java

```diff
@@ -3,88 +3,144 @@
-    }
-    return null;
-  }
-
-  protected Object newInstance(Constructor<?> constructor, Object... initargs) throws InvocationTargetException, InstantiationException, IllegalAccessException {
-    return constructor.newInstance(initargs);
-  }
-
-  @Override
-  public Object methodInvoke(Method method, Object instance, Object... convertedArgs) throws Throwable {
-      try {
-        return method.invoke(instance, convertedArgs);
-      } catch (InvocationTargetException e) {
-        if(e.getCause() != null){
-          throw e.getCause();
-        }else{
-          throw e.getTargetException();
-        }
-      }
-  }
+	@Override
+	public Object createInstance(List<String> paths, String className, Object[] args)
+			throws IllegalArgumentException, InstantiationException,
+			IllegalAccessException, InvocationTargetException {
+		Class<?> k = searchPathsForClass(paths, className);
+		Constructor<?> constructor = getConstructor(k, args);
+		if (constructor == null) {
+			throw new SlimError(String.format("message:<<%s %s>>",
+					SlimServer.NO_CONSTRUCTOR, className));
+		}
+
+		return newInstance(args, constructor);
+	}
+
+	private Object newInstance(Object[] args, Constructor<?> constructor)
+			throws IllegalAccessException, InstantiationException,
+			InvocationTargetException {
+		Object[] initargs = ConverterSupport.convertArgs(args,
+				constructor.getParameterTypes());
+
+		return newInstance(constructor, initargs);
+	}
+
+
+	protected Class<?> searchPathsForClass(List<String> paths, String className) {
+		Class<?> k = getClass(className);
+
+		if (k != null) {
+			return k;
+		}
+		for (String path : paths) {
+			k = getClass(path + "." + className);
+			if (k != null) {
+				return k;
+			}
+		}
+		throw new SlimError(String.format("message:<<%s %s>>", SlimServer.NO_CLASS, className));
+	}
+
+	protected Class<?> getClass(String className) {
+		try {
+			return Class.forName(className);
+		} catch (ClassNotFoundException e) {
+			return null;
+		}
+	}
+
+	protected Constructor<?> getConstructor(Class<?> clazz,
+			Object[] args) {
+
+		for (Constructor<?> constructor : clazz.getConstructors()) {
+			Class<?>[] arguments = constructor.getParameterTypes();
+			if (arguments.length == args.length) {
+				return constructor;
+			}
+		}
+		return null;
+	}
+
+	protected Object newInstance(Constructor<?> constructor, Object... initargs) throws InvocationTargetException, InstantiationException, IllegalAccessException {
+		return constructor.newInstance(initargs);
+	}
+
+	@Override
+	public MethodExecutionResult findAndInvoke(String methodName, Object instance, Object... args) throws Throwable {
+		Method method = findMatchingMethod(methodName, instance.getClass(), args.length);
+		if (method != null) {
+			return this.invokeMethod(instance, method, args);
+		}
+		return MethodExecutionResult.noMethod(methodName, instance.getClass(), args.length);
+	}
+	
+	protected Method findMatchingMethod(String methodName, Class<?> k, int nArgs) {
```

Comment:
Since the signature is being changed, why not supply all args (or at least their types) to this method? 
If an implementation chooses to use only the number of arguments: fine, but that would allow implementing subclasses the option to actually look more closely.